actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -472,6 +473,10 @@ def check_method(name)
       def default_session
         Session.disabled(self)
       end
+
+      def chunked?
+        headers["Transfer-Encoding"] =~ /chunked/i
```

Comment:
Given the [Rack spec](https://github.com/rack/rack/blob/v3.0.4.1/SPEC.rdoc?plain=1#L72) points to the CGI spec [RFC3875 section 4.1.18](https://tools.ietf.org/html/rfc3875#section-4.1.18) for request headers, and it says:

>If multiple header fields with the same field-name are received then the server MUST rewrite them as a single value having the same semantics

we can expect multiple values to be separated by commas, so I understand why you'd go with a Regexp over just a string comparison but it's a bit too lenient. We don't want xxchunkedxx to be allowed.

Another thing is that =~ sets Regexp global variables, and returns the index of the first match. We should use Regexp#match? instead.

Reply:
I will try to improve this Regex to be more strict, but valid at the same time. Thanks for the suggestions!

Reply:
For example the output could be:
```
 Transfer-Encoding: chunked
Transfer-Encoding: compress
Transfer-Encoding: deflate
Transfer-Encoding: gzip

// Several values can be listed, separated by a comma
Transfer-Encoding: gzip, chunked
```
Instead of match how about split and include? There was a recent PR about not using regex match https://github.com/rails/rails/commit/5726b1d1d7665c33830ad114f45258ecf772b7b6

Reply:
Another question, should uppercase CHUNKED entries be supported? It doesn't look like from reading the specification 🤔 .

Before splitting, we should also remove any extra whitespaces from the string, and probably checking for nils, let me give it a shot 🙇

Reply:
We have to deal with whitespace around commas too. So you'll have use a Regexp in the split, it may not give that much of a gain in the end compared to matching once.

Reply:
https://httpwg.org/specs/rfc9112.html#transfer.codings
> All transfer-coding names are case-insensitive and ought to be registered within the HTTP Transfer Coding registry, as defined in [Section 7.3](https://httpwg.org/specs/rfc9112.html#transfer.coding.registry).

Reply:
Besides that, splitting will not catch invalid `Transfer-Encoding` values within the header. I am thinking of applying the following RegEx:

```ruby
/^([a-z]+,\s*)*chunked(,\s*[a-z]+)*$/.match?(headers["Transfer-Encoding"])
```

Tested here: https://regex101.com/r/NS5cm6/1